x-pack/plugins/security_solution/public/detections/components/rules/rule_actions_field/index.tsx

```diff
@@ -120,18 +120,20 @@ export const RuleActionsField: React.FC<Props> = ({
   const setActionParamsProperty = useCallback(
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     (key: string, value: any, index: number) => {
-      const updatedActions = [...actions];
-      updatedActions[index] = {
-        ...updatedActions[index],
-        params: {
-          ...updatedActions[index].params,
-          [key]: value,
-        },
-      };
-      field.setValue(updatedActions);
+      field.setValue((prevValue: RuleAction[]) => {
```

Comment:
Was this needed for a fix, or just removing the actions dependency since it can be retrieved from `setValue`?

Was curious if this ended up being a functional change with leveraging `actions` as the `prevValue` from the field instead of the memoized `actions` since the useMemo has some `isEmpty` initialization:
https://github.com/elastic/kibana/blob/c7262544b54422f8602c57e68a365554cd83f846/x-pack/plugins/security_solution/public/detections/components/rules/rule_actions_field/index.tsx#L89-L92

But testing/debugging locally doesn't seem to show to be any difference. Looks like we could update `setActionIdByIndex` to use this method as well? No change necessary here, just commenting for when we get a chance to refactor things here with the forms lib 🙂

Reply:
This is a FIX. We wanted to update multiple params without the need to nest them under one object eg. `actionParams.message.X`, and `actionParams.message.Y.` Unfortunately calling editAction each time ended up having some race condition and updating just the latest param. Now, we make sure that we have an actual updated value. I don't think this may break anything, but can't test everything so big thanks for your help here! :elasticheart:

Reply:
Just to be more specific, now we update params multiple times and get the results as `actionParams: {x: string, y: string}; No need to have the additional wrapper.